src/Application.php

```diff
@@ -100,6 +103,13 @@ class Application extends MiddlewarePipe implements Router\RouteResultSubjectInt
     private $routes = [];
 
     /**
+     * Internal; factory class for server request (for mocking)
+     *
+     * @var string
+     */
+    private $serverRequestFactory = ServerRequestFactory::class;
```

Comment:
@weierophinney you _can_ mock this using Mockery without this private property + reflection if desired, by the way.

```php
/**
 * @runInSeparateProcess
 * @preserveGlobalState disabled
 */
public function testSomething()
{
    $serverRequestFactory = m::mock('alias:' . ServerRequestFactory::class);
    $serverRequestFactory->shouldReceive('fromGlobals')
        ->once()
        ->andReturn(['foo' => 'bar']);

    $this->assertSame(['foo' => 'bar'], ServerRequestFactory::fromGlobals());
}
```

Both annotations are required for this to work, else you will hit:

```
Mockery\Exception\RuntimeException: Could not load mock Zend\Diactoros\ServerRequestFactory, class already exists
```

No real harm in leaving it how it is now, but IMO the less "this is only used for testing" stuff in real classes, the better.